src/build_loop.rs

```diff
@@ -72,6 +72,7 @@ impl BuildLoop {
     pub fn forever(&mut self) {
         loop {
             let mut go = || -> Result<(), SingleBuildError> {
+                // TODO: real error handling here
```

Comment:
What kind of real error handling? Reading this comment makes me think you're referring to the "waiter exited" expect call. Can you explain a bit more (in the comment) about what this means?

Reply:
Moved the comment a few lines further down and expanded.